spec/models/loan_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'rails_helper'
+
+describe Loan, :type => :model do
+  it_should_behave_like 'translatable', ['Description', 'ShortDescription']
+  it_should_behave_like 'mediable'
+
+  it 'has a valid factory' do
+    expect(create(:loan)).to be_valid
+  end
+
+  context 'model methods' do
+    let(:loan) { create(:loan) }
+
+    describe '.name' do
+      context 'with cooperative' do
+        it 'uses cooperative name' do
+          expect(loan.name).to eq I18n.t :project_with, name: loan.cooperative.name
+        end
+      end
+
+      context 'without cooperative' do
+        let(:loan) { create(:loan, cooperative: nil) }
+        it 'uses project id' do
+          pending('is not currently possible given DB constraints')
+          expect(loan.name).to eq I18n.t :project_id, id: loan.id.to_s
+        end
+      end
+    end
+
+    describe '.country' do
+      it 'gets country' do
+        expect(loan.country).to eq Country.where(name: loan.division.super_division.country).first
```

Comment:
This feels a little too close to the actual implementation. I'd prefer to do something like set the country name when creating the loan and then check the name of the associated country record matches... but actually this will be a straight association in the new structure so it probably won't need to be tested anymore.